network/tcp_test.go

```diff
@@ -363,10 +363,48 @@ func TestTCPConnWithListener(t *testing.T) {
 	}
 }
 
-// will create a TCPListener & open a golang net.TCPConn to it
+// will create a TCPListener with a specific address to listen on &
+// open a golang net.TCPConn to it
+func TestTCPListenerWithListenAddr(t *testing.T) {
+	addr := NewAddress(PlainTCP, "127.0.0.1:5678")
+	listenAddr := NewAddress(PlainTCP, "127.0.0.1:5670")
```

Comment:
Whoa, no no. ListenAddr is not for changing the port number, it is only for changing the IP address in the second argument to net.Listen in network/tcp.go, line 285.

Sorry to deliver this big correction only now...

Reply:
I'm not sure that we don't want to change the listening port, too. In some docker setups you might want to have another 'public' port-#, compared to the listening port-#. So I suggest to keep that functionality. We can add some magic and say that:
1. if there is no `listenAddr`, take the `ServerIdentity.Address`
2. if there is an IP in `listenAddr`, take the port from `ServerIdentity.Address`
3. if there is an IP:Port in `listenAddr`, take this

@ValentinMoullet didn't want to mix IP from `listenAddr` and port from `ServerIdentity.Address`, which is also a valid concern.

Reply:
Exactly. Why would having ListenAddr to change the port be bad @jeffallen ?

For your suggestion @ineiti , I agree that this would make sense, and in the end I'm ok with "mixing" them in the case there is only an IP. All this magic will of course need to be commented.